server/src/main/java/io/druid/server/DruidNode.java

```diff
@@ -128,6 +128,11 @@ public int getPort()
     return port;
   }
 
+  public boolean isListening()
+  {
+    return port >= 0;
```

Comment:
jetty translates 0 as random port and throws exception for -1, which is not consistent with what you've mentioned. I have no idea how to coordinate it.